artemis-selector/src/main/java/org/apache/activemq/artemis/selector/filter/XPathExpression.java

```diff
@@ -79,4 +97,15 @@ public boolean matches(Filterable message) throws FilterException {
       return object == Boolean.TRUE;
    }
 
+   protected static void setupFeatures(DocumentBuilderFactory factory) throws ParserConfigurationException {
+      Properties properties = System.getProperties();
+      for (Map.Entry<Object, Object> prop : properties.entrySet()) {
+         String key = (String) prop.getKey();
+         if (key.startsWith(DOCUMENT_BUILDER_FACTORY_FEATURE)) {
+            String uri = key.split(DOCUMENT_BUILDER_FACTORY_FEATURE + ":")[1];
```

Comment:
Fair enough. FWIW, I took this code from the 5.x code-base where it's been working happily since 2014.

Reply:
Sure, but adding stuff to a fresh codebase is a good time to actually tidy it up and fix things. Rather than introducing it as-is, meaning it likely never happens and folks have to see it and/or possibly hit it later. Essentially, if a trivial bug/improvement is easily spotted in review, and people are being asked to review...

Reply:
I agree, and that's why I fixed it. I fixed some other stuff that was part of copy/paste, but I didn't think this was a problem at the time. I was just clarifying where it came from originally. Thanks for pointing it out.